<a href="https://colab.research.google.com/github/ShivaniTake17/Capstone_Project_EDA/blob/main/EDA_Notebook_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###**Project Name - Global Terrorism Analysis**

####**Project Type** - EDA
####**Contribution** - Individual

# **Project** **Summary**

In this EDA project, we were provided with the Global Terrorism Database (GTD) which is a database including information on terrorist events around the world from 1970 through 2017. Unlike many other event databases, the GTD includes systematic data on domestic as well as international terrorist incidents that have occurred during this time period and now includes more than 181,000 cases.

In this project we are doing the exploratory data analysis on the global terrorisam database to find out various insights of terrorism activity and their effect over region,country and city. Also we did some visualizations for easy understandings of the analysis.

After analyzing the cleaned data, we divided the project into various types of analysis. The first is the analysis based on Trend of Terrorrism Per Year the second is the analysis based on Most affected areas of Terrorism. Third analysis based on Terrorist Organization and the last analysis is based on country.

In the EDA we have conclude that after 2011 there is a large growth recorded in terrorist activity and 2014 appears to be a witness of huge terrorist attack.

# GitHub link -

# Problem Statement

The Global Terrorism Database (GTD) is an open-source database including information on terrorist attacks around the world from 1970 through 2017. The GTD includes systematic data on domestic as well as international terrorist incidents that have occurred during this time period and now includes more than 180,000 attacks. The database is maintained by researchers at the National Consortium for the Study of Terrorism and Responses to Terrorism (START), headquartered at the University of Maryland.

Explore and analyze the data to discover key findings pertaining to terrorist activities

# Objective -

1.Studies in Conflict and Terrorism aims to cast new light on the origins and implications of conflict in the 21st Century and to illuminate new approaches and solutions to countering the growth and escalation of contemporary sub-state violence.

2.The main objective of the project is to find out the effect of terrorism in the world and spread awareness to reduce the violence.

##**Let's Begin !**


In [ ]:
# importing the libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

## Loading The Dataset

In [ ]:
# Mount the drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Location of the datafile
file_path = ('/content/drive/MyDrive/CSV Files/Global Terrorism Data.csv')

In [ ]:
# read the dataset
df=pd.read_csv(file_path,encoding="ISO-8859-1")

<ipython-input-6-fee17133bb24>:2: DtypeWarning: Columns (4,6,31,33,61,62,63,76,79,90,92,94,96,114,115,121) have mixed types. Specify dtype option on import or set low_memory=False.
  df=pd.read_csv(file_path,encoding="ISO-8859-1")


In [ ]:
df.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

In [ ]:
df.tail()

eventid  iyear  imonth  iday approxdate  extended resolution  \
181686  201712310022   2017      12    31        NaN         0        NaN   
181687  201712310029   2017      12    31        NaN         0        NaN   
181688  201712310030   2017      12    31        NaN         0        NaN   
181689  201712310031   2017      12    31        NaN         0        NaN   
181690  201712310032   2017      12    31        NaN         0        NaN   

        country  country_txt  region  ... addnotes  \
181686      182      Somalia      11  ...      NaN   
181687      200        Syria      10  ...      NaN   
181688      160  Philippines       5  ...      NaN   
181689       92        India       6  ...      NaN   
181690      160  Philippines       5  ...      NaN   

                                                   scite1  \
181686  "Somalia: Al-Shabaab Militants Attack Army Che...   
181687  "Putin's 'victory' in Syria has turned into a ...   
181688  "Maguindanao clashes trap tribe members," Phil...   
181689  "Trader escapes grenade attack in Imphal," Bus...   
181690  "Security tightened in Cotabato following IED ...   

                                                   scite2  \
181686  "Highlights: Somalia Daily Media Highlights 2 ...   
181687  "Two Russian soldiers killed at Hmeymim base i...   
181688                                                NaN   
181689                                                NaN   
181690  "Security tightened in Cotabato City," Manila ...   

                                                   scite3  \
181686  "Highlights: Somalia Daily Media Highlights 1 ...   
181687  "Two Russian servicemen killed in Syria mortar...   
181688                                                NaN   
181689                                                NaN   
181690                                                NaN   

                        dbsource  INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
181686  START Primary Collection        0         0        0       0      NaN  
181687  START Primary Collection       -9        -9        1       1      NaN  
181688  START Primary Collection        0         0        0       0      NaN  
181689  START Primary Collection       -9        -9        0      -9      NaN  
181690  START Primary Collection       -9        -9        0      -9      NaN  

[5 rows x 135 columns]

In [ ]:
df.shape

(181691, 135)

In [ ]:
df.size

24528285

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Columns: 135 entries, eventid to related
dtypes: float64(55), int64(22), object(58)
memory usage: 187.1+ MB


In [ ]:
df.describe()

eventid          iyear         imonth           iday  \
count  1.816910e+05  181691.000000  181691.000000  181691.000000   
mean   2.002705e+11    2002.638997       6.467277      15.505644   
std    1.325957e+09      13.259430       3.388303       8.814045   
min    1.970000e+11    1970.000000       0.000000       0.000000   
25%    1.991021e+11    1991.000000       4.000000       8.000000   
50%    2.009022e+11    2009.000000       6.000000      15.000000   
75%    2.014081e+11    2014.000000       9.000000      23.000000   
max    2.017123e+11    2017.000000      12.000000      31.000000   

            extended        country         region       latitude  \
count  181691.000000  181691.000000  181691.000000  177135.000000   
mean        0.045346     131.968501       7.160938      23.498343   
std         0.208063     112.414535       2.933408      18.569242   
min         0.000000       4.000000       1.000000     -53.154613   
25%         0.000000      78.000000       5.000000      11.510046   
50%         0.000000      98.000000       6.000000      31.467463   
75%         0.000000     160.000000      10.000000      34.685087   
max         1.000000    1004.000000      12.000000      74.633553   

          longitude    specificity  ...     ransomamt   ransomamtus  \
count  1.771340e+05  181685.000000  ...  1.350000e+03  5.630000e+02   
mean  -4.586957e+02       1.451452  ...  3.172530e+06  5.784865e+05   
std    2.047790e+05       0.995430  ...  3.021157e+07  7.077924e+06   
min   -8.618590e+07       1.000000  ... -9.900000e+01 -9.900000e+01   
25%    4.545640e+00       1.000000  ...  0.000000e+00  0.000000e+00   
50%    4.324651e+01       1.000000  ...  1.500000e+04  0.000000e+00   
75%    6.871033e+01       1.000000  ...  4.000000e+05  0.000000e+00   
max    1.793667e+02       5.000000  ...  1.000000e+09  1.320000e+08   

         ransompaid  ransompaidus  hostkidoutcome     nreleased  \
count  7.740000e+02    552.000000    10991.000000  10400.000000   
mean   7.179437e+05    240.378623        4.629242    -29.018269   
std    1.014392e+07   2940.967293        2.035360     65.720119   
min   -9.900000e+01    -99.000000        1.000000    -99.000000   
25%   -9.900000e+01      0.000000        2.000000    -99.000000   
50%    0.000000e+00      0.000000        4.000000      0.000000   
75%    1.273412e+03      0.000000        7.000000      1.000000   
max    2.750000e+08  48000.000000        7.000000   2769.000000   

             INT_LOG       INT_IDEO       INT_MISC        INT_ANY  
count  181691.000000  181691.000000  181691.000000  181691.000000  
mean       -4.543731      -4.464398       0.090010      -3.945952  
std         4.543547       4.637152       0.568457       4.691325  
min        -9.000000      -9.000000      -9.000000      -9.000000  
25%        -9.000000      -9.000000       0.000000      -9.000000  
50%        -9.000000      -9.000000       0.000000       0.000000  
75%         0.000000       0.000000       0.000000       0.000000  
max         1.000000       1.000000       1.000000       1.000000  

[8 rows x 77 columns]

In [ ]:
df.columns

Index(['eventid', 'iyear', 'imonth', 'iday', 'approxdate', 'extended',
       'resolution', 'country', 'country_txt', 'region',
       ...
       'addnotes', 'scite1', 'scite2', 'scite3', 'dbsource', 'INT_LOG',
       'INT_IDEO', 'INT_MISC', 'INT_ANY', 'related'],
      dtype='object', length=135)

Checking Duplicate Events

In [ ]:
# check the duplicate event from eventid
df['eventid'].duplicated().sum()

0

# What did you know about your dataset?

In the Global Terrorism Database there are 181691 rows which describe the number of attacks happen between year 1970 to 2017. And there are 135 columns which includes various information about the terrorism.

There is not a single duplicate event in the dataset.

Also there are some null valus in the various column. so we have to replace it with zero for further calculations.


# **Data Wrangling**

###Cleaning and Managing the data

As we dont want to make changes in our original dataframe. We made one copy of the dataset and made the changes that we needed in the copy of the dataset

In [ ]:
new_df=df.copy()

In [ ]:
new_df.shape

(181691, 135)

In [ ]:
new_df.columns

Index(['eventid', 'iyear', 'imonth', 'iday', 'approxdate', 'extended',
       'resolution', 'country', 'country_txt', 'region',
       ...
       'addnotes', 'scite1', 'scite2', 'scite3', 'dbsource', 'INT_LOG',
       'INT_IDEO', 'INT_MISC', 'INT_ANY', 'related'],
      dtype='object', length=135)

In [ ]:
new_df.columns[0:20]

Index(['eventid', 'iyear', 'imonth', 'iday', 'approxdate', 'extended',
       'resolution', 'country', 'country_txt', 'region', 'region_txt',
       'provstate', 'city', 'latitude', 'longitude', 'specificity', 'vicinity',
       'location', 'summary', 'crit1'],
      dtype='object')

In [ ]:
new_df.columns[20:40]

Index(['crit2', 'crit3', 'doubtterr', 'alternative', 'alternative_txt',
       'multiple', 'success', 'suicide', 'attacktype1', 'attacktype1_txt',
       'attacktype2', 'attacktype2_txt', 'attacktype3', 'attacktype3_txt',
       'targtype1', 'targtype1_txt', 'targsubtype1', 'targsubtype1_txt',
       'corp1', 'target1'],
      dtype='object')

In [ ]:
new_df.columns[40:60]

Index(['natlty1', 'natlty1_txt', 'targtype2', 'targtype2_txt', 'targsubtype2',
       'targsubtype2_txt', 'corp2', 'target2', 'natlty2', 'natlty2_txt',
       'targtype3', 'targtype3_txt', 'targsubtype3', 'targsubtype3_txt',
       'corp3', 'target3', 'natlty3', 'natlty3_txt', 'gname', 'gsubname'],
      dtype='object')

In [ ]:
new_df.columns[60:80]

Index(['gname2', 'gsubname2', 'gname3', 'gsubname3', 'motive', 'guncertain1',
       'guncertain2', 'guncertain3', 'individual', 'nperps', 'nperpcap',
       'claimed', 'claimmode', 'claimmode_txt', 'claim2', 'claimmode2',
       'claimmode2_txt', 'claim3', 'claimmode3', 'claimmode3_txt'],
      dtype='object')

In [ ]:
new_df.columns[80:100]

Index(['compclaim', 'weaptype1', 'weaptype1_txt', 'weapsubtype1',
       'weapsubtype1_txt', 'weaptype2', 'weaptype2_txt', 'weapsubtype2',
       'weapsubtype2_txt', 'weaptype3', 'weaptype3_txt', 'weapsubtype3',
       'weapsubtype3_txt', 'weaptype4', 'weaptype4_txt', 'weapsubtype4',
       'weapsubtype4_txt', 'weapdetail', 'nkill', 'nkillus'],
      dtype='object')

In [ ]:
new_df.columns[100:120]

Index(['nkillter', 'nwound', 'nwoundus', 'nwoundte', 'property', 'propextent',
       'propextent_txt', 'propvalue', 'propcomment', 'ishostkid', 'nhostkid',
       'nhostkidus', 'nhours', 'ndays', 'divert', 'kidhijcountry', 'ransom',
       'ransomamt', 'ransomamtus', 'ransompaid'],
      dtype='object')

In [ ]:
new_df.columns[120:136]

Index(['ransompaidus', 'ransomnote', 'hostkidoutcome', 'hostkidoutcome_txt',
       'nreleased', 'addnotes', 'scite1', 'scite2', 'scite3', 'dbsource',
       'INT_LOG', 'INT_IDEO', 'INT_MISC', 'INT_ANY', 'related'],
      dtype='object')

from the above column list theare are many columns which are not needed for the analysis. so we are keeping those columns which are useful to us

Rename those columns which are needed

In [ ]:
# Rename the columns for easy understanding
new_df.rename(columns={'iyear':'Year','imonth':'Month','iday':'Day','country_txt':'Country','city':'City','provstate':'State',
                          'region_txt':'Region','attacktype1_txt':'Attack_Type','targtype1_txt':'Main_Target','nkill':'Killed',
                          'nwound':'Wounded','gname':'Gang_Name',
                          'weaptype1_txt':'Weapon_type','latitude':'Latitude',
                          'longitude':'Longitude','targsubtype1_txt':'Sub_Target',
                          'natlty1_txt':'Nationality','motive':'Motive','summary':'Summary'},inplace=True)

Keep those columns which are important for analysis

In [ ]:
# columns_to_keep=['Year','Month','Day','City','Region','Latitude','Country','State','Longitude','Attack_Type','Killed','Gang_Name','Wounded','Main_Target','Sub_Target','Weapon_type','Motive','Nationality']

In [ ]:
new_df=new_df[['Year','Month','Day','Latitude','Longitude','Region','Country','State','City','Attack_Type','Killed','Gang_Name','Wounded','Main_Target','Sub_Target','Weapon_type','Motive','Nationality']]

###**Understanding the variable**

In [ ]:
new_df.columns

Index(['Year', 'Month', 'Day', 'Latitude', 'Longitude', 'Region', 'Country',
       'State', 'City', 'Attack_Type', 'Killed', 'Gang_Name', 'Wounded',
       'Main_Target', 'Sub_Target', 'Weapon_type', 'Motive', 'Nationality'],
      dtype='object')

In [ ]:
new_df.shape

(181691, 18)

In [ ]:
new_df.describe()

Year          Month            Day       Latitude  \
count  181691.000000  181691.000000  181691.000000  177135.000000   
mean     2002.638997       6.467277      15.505644      23.498343   
std        13.259430       3.388303       8.814045      18.569242   
min      1970.000000       0.000000       0.000000     -53.154613   
25%      1991.000000       4.000000       8.000000      11.510046   
50%      2009.000000       6.000000      15.000000      31.467463   
75%      2014.000000       9.000000      23.000000      34.685087   
max      2017.000000      12.000000      31.000000      74.633553   

          Longitude         Killed        Wounded  
count  1.771340e+05  171378.000000  165380.000000  
mean  -4.586957e+02       2.403272       3.167668  
std    2.047790e+05      11.545741      35.949392  
min   -8.618590e+07       0.000000       0.000000  
25%    4.545640e+00       0.000000       0.000000  
50%    4.324651e+01       0.000000       0.000000  
75%    6.871033e+01       2.000000       2.000000  
max    1.793667e+02    1570.000000    8191.000000

In [ ]:
new_df.head(2)

Year  Month  Day   Latitude  Longitude                       Region  \
0  1970      7    2  18.456792 -69.951164  Central America & Caribbean   
1  1970      0    0  19.371887 -99.086624                North America   

              Country    State           City                  Attack_Type  \
0  Dominican Republic      NaN  Santo Domingo                Assassination   
1              Mexico  Federal    Mexico city  Hostage Taking (Kidnapping)   

   Killed                           Gang_Name  Wounded  \
0     1.0                              MANO-D      0.0   
1     0.0  23rd of September Communist League      0.0   

                   Main_Target  \
0  Private Citizens & Property   
1      Government (Diplomatic)   

                                          Sub_Target Weapon_type Motive  \
0                                     Named Civilian     Unknown    NaN   
1  Diplomatic Personnel (outside of embassy, cons...     Unknown    NaN   

          Nationality  
0  Dominican Republic  
1             Belgium

In [ ]:
new_df.tail(2)

Year  Month  Day   Latitude   Longitude          Region      Country  \
181689  2017     12   31  24.798346   93.940430      South Asia        India   
181690  2017     12   31   7.209594  124.241966  Southeast Asia  Philippines   

              State           City        Attack_Type  Killed Gang_Name  \
181689      Manipur         Imphal  Bombing/Explosion     0.0   Unknown   
181690  Maguindanao  Cotabato City  Bombing/Explosion     0.0   Unknown   

        Wounded           Main_Target                           Sub_Target  \
181689      0.0  Government (General)  Government Building/Facility/Office   
181690      0.0               Unknown                                  NaN   

       Weapon_type Motive  Nationality  
181689  Explosives    NaN        India  
181690  Explosives    NaN  Philippines

###**Variable Description**

Year :- Year of Terrorist Attack

Month :- Months of Terrorist Attack

Day :- On which Day Terrorist Attack is happend

Latitude :- Y Co-ordinates of Terrorist Attack point

Longitude :- X Co-ordinate of Terrorist Attack point

Region :- Regions in which Terrorist Attack happens

Country :- Countries in which Terrorist Attack happens

State :- States in which Terrorist Attack happens

City :- Cities in which Terrorist Attack happens

Attack_Type :- Type of Terrorist Attack

Killed :- Number of people killed in the Terrorist Attack

Gang_Name :- Name of Terrorist Organization

Wounded :- Number of people Injured in the Terrorist Attack

Main_Target :- Main taget of the Terrorist Attack

Sub_Target :- Sub target of Terrorist Attack

Weopan_type :- Weopan used in the Terrorist Attack

Motive :- Motive of Terrorist Attack

Nationality :- Nationality of Terrorist Organization

Casualties :- It is the Sum of Number of people killed and Number  of people Injured in the Terrorist Attack

In [ ]:
new_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Data columns (total 18 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   Year         181691 non-null  int64  
 1   Month        181691 non-null  int64  
 2   Day          181691 non-null  int64  
 3   Latitude     177135 non-null  float64
 4   Longitude    177134 non-null  float64
 5   Region       181691 non-null  object 
 6   Country      181691 non-null  object 
 7   State        181270 non-null  object 
 8   City         181256 non-null  object 
 9   Attack_Type  181691 non-null  object 
 10  Killed       171378 non-null  float64
 11  Gang_Name    181691 non-null  object 
 12  Wounded      165380 non-null  float64
 13  Main_Target  181691 non-null  object 
 14  Sub_Target   171318 non-null  object 
 15  Weapon_type  181691 non-null  object 
 16  Motive       50561 non-null   object 
 17  Nationality  180132 non-null  object 
dtypes: float64(4), int64(3),

In [ ]:
#checking the total number of null values of the columns
new_df.isnull().sum()

Year                0
Month               0
Day                 0
Latitude         4556
Longitude        4557
Region              0
Country             0
State             421
City              435
Attack_Type         0
Killed          10313
Gang_Name           0
Wounded         16311
Main_Target         0
Sub_Target      10373
Weapon_type         0
Motive         131130
Nationality      1559
dtype: int64

From the above data, We can see that there are many columns which has null values

There is no need to remove null values of all the columns. instead we are replacing the null values of killed and wounded column with zero

In [ ]:
# fetch the null value of killed column
new_df[new_df['Killed'].isnull()].head(2)

Year  Month  Day   Latitude   Longitude          Region Country    State  \
3  1970      1    0  37.997490   23.762728  Western Europe  Greece   Attica   
4  1970      1    0  33.580412  130.396361       East Asia   Japan  Fukouka   

      City                     Attack_Type  Killed Gang_Name  Wounded  \
3   Athens               Bombing/Explosion     NaN   Unknown      NaN   
4  Fukouka  Facility/Infrastructure Attack     NaN   Unknown      NaN   

               Main_Target         Sub_Target Weapon_type Motive  \
3  Government (Diplomatic)  Embassy/Consulate  Explosives    NaN   
4  Government (Diplomatic)  Embassy/Consulate  Incendiary    NaN   

     Nationality  
3  United States  
4  United States

In [ ]:
# fetch the null value of Wounded column
new_df[new_df['Wounded'].isnull()].head(2)

Year  Month  Day   Latitude   Longitude          Region Country    State  \
3  1970      1    0  37.997490   23.762728  Western Europe  Greece   Attica   
4  1970      1    0  33.580412  130.396361       East Asia   Japan  Fukouka   

      City                     Attack_Type  Killed Gang_Name  Wounded  \
3   Athens               Bombing/Explosion     NaN   Unknown      NaN   
4  Fukouka  Facility/Infrastructure Attack     NaN   Unknown      NaN   

               Main_Target         Sub_Target Weapon_type Motive  \
3  Government (Diplomatic)  Embassy/Consulate  Explosives    NaN   
4  Government (Diplomatic)  Embassy/Consulate  Incendiary    NaN   

     Nationality  
3  United States  
4  United States

In [ ]:
# replace all the null value of killed column with 0
new_df.loc[new_df['Killed'].isna(),'Killed']= 0

In [ ]:
# replace all the null value of Wounded column with 0
new_df.loc[new_df['Wounded'].isna(),'Wounded']= 0

In [ ]:
# checking the replaced value
new_df.loc[3:4,'Killed':'Wounded']

Killed Gang_Name  Wounded
3     0.0   Unknown      0.0
4     0.0   Unknown      0.0

Successfully replace the null value of killed and wounded column with zero

In [ ]:
# now lets create one column named Casualties by using killed and wounded column

In [ ]:
new_df['Casualties']=new_df['Killed']+new_df['Wounded']

In [ ]:
new_df.shape

(181691, 19)

In [ ]:
new_df.head(2)

Year  Month  Day   Latitude  Longitude                       Region  \
0  1970      7    2  18.456792 -69.951164  Central America & Caribbean   
1  1970      0    0  19.371887 -99.086624                North America   

              Country    State           City                  Attack_Type  \
0  Dominican Republic      NaN  Santo Domingo                Assassination   
1              Mexico  Federal    Mexico city  Hostage Taking (Kidnapping)   

   Killed                           Gang_Name  Wounded  \
0     1.0                              MANO-D      0.0   
1     0.0  23rd of September Communist League      0.0   

                   Main_Target  \
0  Private Citizens & Property   
1      Government (Diplomatic)   

                                          Sub_Target Weapon_type Motive  \
0                                     Named Civilian     Unknown    NaN   
1  Diplomatic Personnel (outside of embassy, cons...     Unknown    NaN   

          Nationality  Casualties  
0  Dominican Republic         1.0  
1             Belgium         0.0

In [ ]:
new_df.describe()

Year          Month            Day       Latitude  \
count  181691.000000  181691.000000  181691.000000  177135.000000   
mean     2002.638997       6.467277      15.505644      23.498343   
std        13.259430       3.388303       8.814045      18.569242   
min      1970.000000       0.000000       0.000000     -53.154613   
25%      1991.000000       4.000000       8.000000      11.510046   
50%      2009.000000       6.000000      15.000000      31.467463   
75%      2014.000000       9.000000      23.000000      34.685087   
max      2017.000000      12.000000      31.000000      74.633553   

          Longitude         Killed        Wounded     Casualties  
count  1.771340e+05  181691.000000  181691.000000  181691.000000  
mean  -4.586957e+02       2.266860       2.883296       5.150156  
std    2.047790e+05      11.227057      34.309747      40.555416  
min   -8.618590e+07       0.000000       0.000000       0.000000  
25%    4.545640e+00       0.000000       0.000000       0.000000  
50%    4.324651e+01       0.000000       0.000000       1.000000  
75%    6.871033e+01       2.000000       2.000000       4.000000  
max    1.793667e+02    1570.000000    8191.000000    9574.000000

##Analysis and visualization of data

###Number of Terrorist Attacks per year/ Yearly growth of terrorist attacks.


In [ ]:
# number of attacks per year.
Attacks_per_year=new_df['Year'].value_counts()
Attacks_per_year

Year
2014    16903
2015    14965
2016    13587
2013    12036
2017    10900
2012     8522
2011     5076
1992     5071
2010     4826
2008     4805
2009     4721
1991     4683
1989     4324
1990     3887
1988     3721
1984     3495
1994     3456
2007     3242
1997     3197
1987     3183
1995     3081
1996     3058
1985     2915
1983     2870
1986     2860
2006     2758
1979     2662
1980     2662
1981     2586
1982     2544
2005     2017
2001     1906
2000     1814
1978     1526
1999     1395
2002     1333
1977     1319
2003     1278
2004     1166
1998      934
1976      923
1975      740
1970      651
1974      581
1972      568
1973      473
1971      471
Name: count, dtype: int64

In [ ]:
plt.figure(figsize = (18,6))
sns.lineplot(Attacks_per_year.index, Attacks_per_year.values)
plt.title('Number of Terrorist Attacks per year',fontsize=20)
plt.xlabel('Years',fontsize=15)
plt.ylabel('Number of attacks',fontsize=15)
plt.show()

TypeError: lineplot() takes from 0 to 1 positional arguments but 2 were given

<Figure size 1800x600 with 0 Axes>